src/stdlib/io/file.rs

```diff
@@ -0,0 +1,323 @@
+use crate::{Context, Error, IntoValue, Value};
+use std::{
+    cell::RefCell,
+    fs::File,
+    io::{self, Cursor, Read, Seek, Write},
+    rc::Rc,
+};
+
+use super::std_file_kind::StdFileKind;
+use either::Either;
+use gc_arena::Collect;
+
+fn read_from_any<'gc, W: Write + Read + Seek>(
+    ctx: Context<'gc>,
+    file: &mut W,
+    format: &str,
```

Comment:
It would be good to make the format an enum, so that this function doesn't need to deal with parsing the format specifier.